src/video_core/rasterizer_accelerated.cpp

```diff
@@ -103,12 +104,19 @@ RasterizerAccelerated::VertexArrayInfo RasterizerAccelerated::AnalyzeVertexArray
 
         vertex_min = 0xFFFF;
         vertex_max = 0;
-        const u32 size = regs.pipeline.num_vertices * (index_u16 ? 2 : 1);
+        const u32 count = regs.pipeline.num_vertices;
+        const u32 index_size = index_u16 ? 2 : 1;
+        const u32 size = count * index_size;
         FlushRegion(address, size);
-        for (u32 index = 0; index < regs.pipeline.num_vertices; ++index) {
-            const u32 vertex = index_u16 ? index_address_16[index] : index_address_8[index];
-            vertex_min = std::min(vertex_min, vertex);
-            vertex_max = std::max(vertex_max, vertex);
+
+        if (index_u16) {
+            const auto res = Common::FindMinMax({index_address_16, static_cast<size_t>(count)});
+            vertex_min = static_cast<u32>(res.first);
+            vertex_max = static_cast<u32>(res.second);
+        } else {
+            const auto res = Common::FindMinMax({index_address_8, static_cast<size_t>(count)});
+            vertex_min = static_cast<u32>(res.first);
+            vertex_max = static_cast<u32>(res.second);
```

Comment:
The last two lines can be deduplicated

Reply:
They can't, res is a different type on each block